File: Arrays and Strings/392. Is Subsequence/Question_4.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>


bool isSubsequence(char* s, char*t) {
    int i =0, j=0;
    while (i < strlen(s) && j < strlen(t)) {

        if(s[i] == t[j]){
            i++;
            j++;
        }else{
            j++;
        }
    }
    return i == strlen(s);
    
}

int main() {
    char *s = "abc";
    char *t = "ahbgdc";

    printf("%d", isSubsequence(s,t));

    return 0;
}

File: Arrays and Strings/392. Is Subsequence/Question_4.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int i =0, j=0;
        while (i < s.length() && j < t.length()) {

            if(s.charAt(i) == t.charAt(j)){
                i++;
                j++;
            }else{
                j++;
            }
        }
        return i == s.length();
    }
}
public class Question_4 {
    public static void main(String args[]){
        Solution sol = new Solution();
        String s = "abc";
        String t = "awnbghc";
    
        System.out.println(sol.isSubsequence(s,t));
        System.exit(0);
    }

}
